captum/attr/_core/guided_backprop_deconvnet.py

```diff
@@ -2,29 +2,41 @@
 import warnings
 import torch
 import torch.nn.functional as F
+from typing import Callable, Any, List, Union, Tuple, Optional
+
+from torch import Tensor
+from torch.nn import Module
+from torch.utils.hooks import RemovableHandle
 
 from .._utils.attribution import GradientAttribution
 from .._utils.common import _format_input, _format_attributions
 from .._utils.gradient import apply_gradient_requirements, undo_gradient_requirements
+from .._utils.typing import TensorOrTupleOfTensors
 
 
 class ModifiedReluGradientAttribution(GradientAttribution):
-    def __init__(self, model, use_relu_grad_output=False):
+    def __init__(self, model: Module, use_relu_grad_output: bool = False):
         r"""
         Args:
 
             model (nn.Module):  The reference to PyTorch model instance.
         """
         GradientAttribution.__init__(self, model)
         self.model = model
-        self.backward_hooks = []
+        self.backward_hooks: List[RemovableHandle] = []
         self.use_relu_grad_output = use_relu_grad_output
         assert isinstance(self.model, torch.nn.Module), (
             "Given model must be an instance of torch.nn.Module to properly hook"
             " ReLU layers."
         )
 
-    def attribute(self, inputs, target=None, additional_forward_args=None):
+    def attribute(
+            self,
+            inputs: TensorOrTupleOfTensors,
+            target: Optional[
+                Union[int, Tuple[int, ...], Tensor, List[Tuple[int, ...]]]
+            ] = None,
+            additional_forward_args: Any = None):
```

Comment:
This should return TensorOrTupleOfTensors, and similarly for GuidedBackprop and Deconvolution

Reply:
done